File: Sean的leetcode/Leetcode/Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string res;
        if (strs.size() == 0) return res;

        for (int i = 0; i < strs[0].size(); i++)
        {
            char c = strs[0][i];
            int j;
            for (j = 1; j < strs.size(); j++)
            {
                if (i >= strs[j].size() || strs[j][i] != c)
                    break;
            }
            if (j == strs.size())
                res += c;
            else
                break;
        }
        return res;
    }
};

File: MyLeetCode/python/Longest Common Prefix.py
###
# Vertical Scan
# Time Complexity: O(n)
# Space Complexity: O(1)
###
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ""
        end = 0
        for i in xrange(len(strs[0])):
            for j in xrange(1, len(strs)):
                if i == len(strs[j]) or strs[j][i] != strs[0][i]:
                    return strs[0][:i]
            
        return strs[0]
                                        
                            
                    